src/GitReleaseManager.Core/Provider/GitLabProvider.cs

```diff
@@ -0,0 +1,435 @@
+
+        public Task CreateLabelAsync(string owner, string repository, Label label)
+        {
+            // The label functionality in GitLab already provides more than
+            // what is possible in GRM. As such, the decision was taken to not
+            // implement the creation of labels for the GitLab provider.
+            throw new NotImplementedException();
+        }
+
+        public Task DeleteLabelAsync(string owner, string repository, Label label)
+        {
+            // The label functionality in GitLab already provides more than
+            // what is possible in GRM. As such, the decision was taken to not
+            // implement the deletion of labels for the GitLab provider.
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<Label>> GetLabelsAsync(string owner, string repository)
+        {
+            // The label functionality in GitLab already provides more than
+            // what is possible in GRM. As such, the decision was taken to not
+            // implement the fetching of labels for the GitLab provider.
+            throw new NotImplementedException();
+        }
+
+        public Task<Milestone> GetMilestoneAsync(string owner, string repository, string milestoneTitle, ItemStateFilter itemStateFilter = ItemStateFilter.All)
+        {
+            return ExecuteAsync(async () =>
+            {
+                var milestones = await GetMilestonesAsync(owner, repository, itemStateFilter).ConfigureAwait(false);
+                var foundMilestone = milestones.FirstOrDefault(m => m.Title == milestoneTitle);
+
+                if (foundMilestone is null)
+                {
+                    throw new NotFoundException(NOT_FOUND_MESSGAE);
+                }
+
+                return foundMilestone;
+            });
+        }
+
+        public Task<IEnumerable<Milestone>> GetMilestonesAsync(string owner, string repository, ItemStateFilter itemStateFilter = ItemStateFilter.All)
+        {
+            return ExecuteAsync(async () =>
+            {
+                var query = new MilestoneQuery();
+
+                if (itemStateFilter == ItemStateFilter.Open)
+                {
+                    query.State = MilestoneState.active;
+                }
+                else if (itemStateFilter == ItemStateFilter.Closed)
+                {
+                    query.State = MilestoneState.closed;
+                }
+
+                var mileStoneClient = _gitLabClient.GetMilestone(GetGitLabProjectId(owner, repository));
+
+                var milestones = mileStoneClient.Get(query);
+                var mappedMilestones = _mapper.Map<IEnumerable<Milestone>>(milestones);
+
+                foreach (var mappedMilestone in mappedMilestones)
+                {
+                    mappedMilestone.HtmlUrl = string.Format(CultureInfo.InvariantCulture, "https://gitlab.com/{0}/{1}/-/milestones/{2}#tab-issues", owner, repository, mappedMilestone.PublicNumber);
```

Comment:
Can we extract this magic string format to a constant at the top of the file (or maybe into a different static class)?